packages/web/tests/components/DiscordCheck/DiscordButtonCheck.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import fetchMock from 'fetch-mock-jest';
+import userEvent from '@testing-library/user-event';
+import { DiscordButtonCheck } from '../../../src/components/DiscordCheck';
+import { act, render, screen } from '../../testUtils/testTools';
+
+jest.mock('next/router', () => ({
+  useRouter() {
+    return {
+      query: '',
+    };
+  },
+}));
+
+const useRouter = jest.spyOn(require('next/router'), 'useRouter');
+
+interface User {
+  discordId?: string;
+}
+
+const sampleUser: User = {
+  discordId: undefined,
+};
+
+const sampleUser2: User = {
+  discordId: '34523452345',
+};
+
+const wait = () => new Promise<void>((resolve) => setTimeout(() => resolve(), 0));
+
+describe('check for discordId to change button', () => {
```

Comment:
The describe should _**describe**_ what's being tested, not the outcome
```suggestion
describe('DiscordButton component', () => {
```